hummingbot/connector/gateway/amm/gateway_amm_base.py

```diff
@@ -0,0 +1,584 @@
+            "account_balance": len(self._account_balances) > 0 if self._trading_required else True,
+            "native_currency": self._native_currency is not None,
+            "network_transaction_fee": self.network_transaction_fee is not None if self._trading_required else True,
+        }
+
+    async def check_network(self) -> NetworkStatus:
+        """
+        Checks if the gateway is up and running.
+        """
+        try:
+            if await self._get_gateway_instance().ping_gateway():
+                return NetworkStatus.CONNECTED
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            return NetworkStatus.NOT_CONNECTED
+        return NetworkStatus.NOT_CONNECTED
+
+    def tick(self, timestamp: float):
+        """
+        Is called automatically by the clock for each clock's tick (1 second by default).
+        It checks if status polling task is due for execution.
+        """
+        if time.time() - self._last_poll_timestamp > self.POLL_INTERVAL:
+            if self._poll_notifier is not None and not self._poll_notifier.is_set():
+                self._poll_notifier.set()
+
+    async def update_balances(self, on_interval: bool = False):
+        """
+        Calls Solana API to update total and available balances.
+        """
+        if self._native_currency is None:
+            await self.get_chain_info()
+        connector_tokens = GatewayConnectionSetting.get_connector_spec_from_market_name(self._name).get("tokens", "").split(",")
+        last_tick = self._last_balance_poll_timestamp
+        current_tick = self.current_timestamp
+        if not on_interval or (current_tick - last_tick) > self.UPDATE_BALANCE_INTERVAL:
+            self._last_balance_poll_timestamp = current_tick
+            local_asset_names = set(self._account_balances.keys())
+            remote_asset_names = set()
+            token_list = list(self._tokens) + [self._native_currency] + connector_tokens
+            resp_json: Dict[str, Any] = await self._get_gateway_instance().get_balances(
+                chain=self.chain,
+                network=self.network,
+                address=self.address,
+                token_symbols=token_list
+            )
+            for token, bal in resp_json["balances"].items():
+                self._account_available_balances[token] = Decimal(str(bal))
+                self._account_balances[token] = Decimal(str(bal))
+                remote_asset_names.add(token)
+            asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+            for asset_name in asset_names_to_remove:
+                del self._account_available_balances[asset_name]
+                del self._account_balances[asset_name]
+            self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._order_tracker.all_orders.items()}
+            self._in_flight_orders_snapshot_timestamp = self.current_timestamp
+
+    async def _update_balances(self):
```

Comment:
maybe is not necessary 2 methods that do the same thing, usually the public method should call the private one